sky/packages/sky/lib/src/fn3/scrollable.dart

```diff
@@ -207,8 +207,56 @@ abstract class ScrollableState<T extends Scrollable> extends State<T> {
   }
 }
 
-// TODO(abarth): findScrollableAncestor
-// TODO(abarth): ensureWidgetIsVisible
+ScrollableState findScrollableAncestor(BuildContext context) {
+  ScrollableState result;
+  context.visitAncestorElements((Element element) {
+    if (element is StatefulComponentElement) {
+      if (element.state is ScrollableState) {
+        result = element.state;
+        return false;
+      }
+    }
+    return true;
+  });
+  return result;
+}
+
+Future ensureWidgetIsVisible(BuildContext context, { Duration duration, Curve curve }) {
+  assert(context.findRenderObject() is RenderBox);
+
+  ScrollableState scrollable = findScrollableAncestor(context);
+  if (scrollable == null)
+    return new Future.value();
+
+  RenderBox targetRenderObject = context.findRenderObject();
+  assert(targetRenderObject.attached);
+  Size targetSize = targetRenderObject.size;
+  Point targetCenter = targetRenderObject.localToGlobal(
+    scrollable.config.scrollDirection == ScrollDirection.vertical
+      ? new Point(0.0, targetSize.height / 2.0)
+      : new Point(targetSize.width / 2.0, 0.0)
+  );
+
+  RenderBox scrollableRenderObject = scrollable.context.findRenderObject();
+  assert(scrollableRenderObject.attached);
+  Size scrollableSize = scrollableRenderObject.size;
+  Point scrollableCenter = scrollableRenderObject.localToGlobal(
+    scrollable.config.scrollDirection == ScrollDirection.vertical
+      ? new Point(0.0, scrollableSize.height / 2.0)
+      : new Point(scrollableSize.width / 2.0, 0.0)
+  );
+  double scrollOffsetDelta = scrollable.config.scrollDirection == ScrollDirection.vertical
+    ? targetCenter.y - scrollableCenter.y
+    : targetCenter.x - scrollableCenter.x;
+  ExtentScrollBehavior scrollBehavior = scrollable.scrollBehavior;
+  double scrollOffset = (scrollable.scrollOffset + scrollOffsetDelta)
+    .clamp(scrollBehavior.minScrollOffset, scrollBehavior.maxScrollOffset);
+
+  if (scrollOffset != scrollable.scrollOffset)
+    return scrollable.scrollTo(scrollOffset, duration: duration, curve: curve);
+
```

Comment:
this doesn't handle nested scrollables. Either assert that there aren't nested scrollables, or add a TODO and/or file a bug for that case.